### DISCUSSION TITLE
convert -extent CMYK tiff result in grayscale file

### DISCUSSION THREAD

**Discussion by qing917** [ImageMagick/ImageMagick#2611] [Help]
Hi there,

I'm new here, would appreciate if someone can help to provide some pointer on this issue I have after moving from 6.7.8.9-18 to current 6.9.10.68-3

Previously with 6.7.8.9-18, I can extent a 1x1 (1 pixel) CMYK tiff file into 3x1 (3 pixel) CMYK tiff file using command, and filled with cmyk(0,0,0,0) for the new pixels using:
`convert test1x1.tif -gravity west -extent 3x1+0+0 test3x1.tif`

But now, it produce a all black 3 pixel grayscale file.
Source file:
`
$ identify test1x1.tif 
test1x1.tif TIFF 1x1 1x1+0+0 8-bit CMYK 348B 0.000u 0:00.001`
Actual output file using 6.9.10.68-3:
`
$ identify test3x1.tif 
test3x1.tif TIFF 3x1 3x1+0+0 8-bit Grayscale Gray 328B 0.000u 0:00.001
`
Expected output file using 6.7.8.9-18:
`
sh-4.2$ identify test3x1.tif 
test3x1.tif TIFF 3x1 3x1+0+0 8-bit CMYK 370B 0.000u 0:00.001`

I'm having issue to downgrade imagemagick on my system, and wonder how should I fix my script to suit the new imageMagick.  Thanks!

**Comment by snibgo** [comment]
I don't know why that no longer works. Perhaps a recent bug. Setting a CMYK background doesn't help. A workaround seems to be inserting "-set colorspace CMYK" after the "-extent".

**Reply by qing917** [reply]
Thanks @snibgo.   "inserting "-set colorspace CMYK" after the "-extent"." helps to get a CMYK output file, but it modified my original single pixel cmyk(0, 0, 0, 100) to some random color values looks greyish in the 3-pixel output file, i.e. [cmyk(0,0,0,42), cmyk(0,0,0,18), cmyk(0,0,0,50)].  Old imageMagick output 3 pixel to be [cmyk(0, 0, 0, 100), cmyk(0, 0, 0, 0), cmyk(0, 0, 0, 0)] .

How can I report this bug if possible?

**Comment by snibgo** [comment]
Post a reproducible example to https://github.com/ImageMagick/ImageMagick/issues

**Reply by dlemstra** [reply]
Please also include the 1x1 test image or instructions on how to create it.

**Reply by qing917** [reply]
Thanks @snibgo and @dlemstra .  Will do.

**Comment by qing917** [comment]
Just to update, I've managed to downgrade to 6.7.8.9-15, it works well for my use case.  I'm happy to close this discussion if there's such a practice.  Thanks for the help.
